astroplan/constraints.py

```diff
@@ -0,0 +1,565 @@
+    Abstract class for objects defining observational constraints.
+    """
+    __metaclass__ = ABCMeta
+
+    def __call__(self, time_range, observer, targets,
+                 time_resolution=DEFAULT_TIME_RESOLUTION):
+
+        if not isinstance(time_range, Time):
+            time_range = Time(time_range)
+        cons = self._compute_constraint(time_range, observer, targets,
+                                        time_resolution=time_resolution)
+        return cons
+
+    def _compute_constraint(self, time_range, observer, targets):
+        # Should be implemented on each subclass of Constraint
+        raise NotImplementedError
+
+    def _get_altaz(self, time_range, observer, targets,
+                   time_resolution=DEFAULT_TIME_RESOLUTION,
+                   force_zero_pressure=False):
+        """
+        Calculate alt/az for ``target`` at times linearly spaced between
+        the two times in ``time_range`` with grid spacing ``time_resolution``
+        for ``observer``.
+
+        Cache the result on the ``observer`` object.
+
+        Parameters
+        ----------
+        time_range : `~astropy.time.Time`
+            Lower and upper time bounds on which to compute the constraints.
+            If input is a scalar time, will only compute at that time (no
+            range).
+
+        targets : {list, `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`}
+            Target or list of targets
+
+        observer : `~astroplan.Observer`
+            The observer who has constraints ``constraints``
+
+        time_resolution : `~astropy.units.Quantity` (optional)
+            Set the time resolution in calculations of the altitude/azimuth
+
+        Returns
+        -------
+        altaz_dict : dict
+            Dictionary containing two key-value pairs. (1) 'times' contains the
+            times for the alt/az computations, (2) 'altaz' contains the
+            corresponding alt/az coordinates at those times.
+        """
+        if not hasattr(observer, '_altaz_cache'):
+            observer._altaz_cache = {}
+
+        times = time_grid_from_range(time_range,
+                                     time_resolution=time_resolution)
+
+        # convert times, targets to tuple for hashing
+        aakey = (tuple(times.jd), tuple(targets))
+
+        if aakey not in observer._altaz_cache:
+            if force_zero_pressure:
+                observer_old_pressure = observer.pressure
+                observer.pressure = 0
+
+            altaz = observer.altaz(times, targets)
+            observer._altaz_cache[aakey] = dict(times=times,
+                                                altaz=altaz)
+            if force_zero_pressure:
+                observer.pressure = observer_old_pressure
+
+        return observer._altaz_cache[aakey]
+
+class AltitudeConstraint(Constraint):
+    """
+    Constrain the altitude of the target.
+
+    Note: this will misbehave if you try to constrain negative altitudes, as
```

Comment:
Instead of "Note:", I suggest using:
```
.. note:: 
    This will misbehave ...
```
If you do that, the "note" will appear much prettier in the sphinx documentation. (Same comment below for all the places you do "note:")